src/include/duckdb/function/function.hpp

```diff
@@ -47,9 +47,13 @@ enum class FunctionStability : uint8_t { CONSISTENT = 0, VOLATILE = 1, CONSISTEN
 struct FunctionData {
 	DUCKDB_API virtual ~FunctionData();
 
+	DUCKDB_API FunctionData();
+	FunctionData(const FunctionData &other);
 	DUCKDB_API virtual unique_ptr<FunctionData> Copy() const = 0;
 	DUCKDB_API virtual bool Equals(const FunctionData &other) const = 0;
 	DUCKDB_API static bool Equals(const FunctionData *left, const FunctionData *right);
+	bool with_ordinality = false;
```

Comment:
`FunctionData` is a very generic struct, also used for scalar and aggregate functions. Could we move this out of here?